ops/ccs-ops-misc/load_test/common/data.py

```diff
@@ -1,60 +1,43 @@
 '''Set up data for use in the tests.'''
 
+from argparse import Namespace
+import logging
 from typing import Callable, List
 import datetime
-import os
 
-from . import config, test_setup as setup
-
-
-def load_all(load_function: Callable, *args, use_table_sample: bool = False) -> List:
-    '''Loads all of the data from the database, using the database connection from the
-    configuration file.'''
-
-    if setup.is_master_thread():
-        ## Don't bother loading data for the master thread, it doenst run a test
+def load_from_parsed_opts(parsed_opts: Namespace, load_function: Callable, *args, use_table_sample: bool = False) -> List:
+    """Loads data from the database given the database load function provided. Gets the database URI and the
+    table sampling percent from the given parsed options. Returns an empty list if database URI and/or table sample percent are not
+    set in the given parsed options
+
+    Args:
+        parsed_opts (Namespace): A collection of parsed options that includes the database URI and table sampling percentage
+        load_function (Callable): A database load function that will query the database to get the desired data
+        use_table_sample (bool, optional): Whether or not to use Postgres's table sampling to randomly sample a given table's data. Defaults to False.
+
+    Returns:
+        List: A list of data returned by the load function
+    """
+    if not parsed_opts.database_uri:
+        logging.getLogger().error('"database_uri" was not defined in parsed options')
         return []
 
-    config_file = config.load()
+    database_uri = str(parsed_opts.database_uri)
+    table_sample_percent = float(parsed_opts.table_sample_percent)
+    return load_from_uri(database_uri, load_function, *args, use_table_sample=use_table_sample, table_sample_percent=table_sample_percent)
+
+def load_from_uri(database_uri: str, load_function: Callable, *args, use_table_sample: bool = False, table_sample_percent: float = 0.25) -> List:
+    '''Loads all of the data from the database, using the database connection provided.'''
```

Comment:
Yeah, I was thinking about this yesterday and I believe that the way we're loading data should probably go through another round of revisions. As it is now, it's OK for the regression suite, but what we really want is per-Locustfile data management instead of the pseudo-OOP we've got going on where these "abstract" base `HttpUser`s manage loading the data and the suites just import the `Task`s from them.

Here's what I'm thinking would be most ideal for future work, perhaps either for the Load Tests or for some distinct data loading/technical debt card:

* Remove all of the single endpoint test suites, and instead use `@tag`s to allow for users to pick and choose which tasks from a particular suite, regression or load, they want to run
  * Perhaps even remove the "abstract" base classes as well (`BeneTestUser`, etc.) -- they're a good idea as far as reducing code duplication, but I'm starting to lean more towards the opinion that test code should be as under-engineered as possible. Particularly, test code should be tailored to the test case they're testing, and it's especially hard to do that with respect to data when we have this inheritance going on
* Make the data loading tailored to each Locustfile/test suite. The regression suite will have its own way of loading data (a consistent set, loaded once during `init` that's distributed to each `HttpUser`) and the load suite will have its own way (table sampled set, loaded per-`HttpUser`). Right now it'll be messy to that, as we'll have to pick one way of being the "default" (due to the abstract classes handling data load) and add some sort of flag or switch to turn it off so that we're not wasting time loading data that's not used.